src/bin/lttng-sessiond/save.c

```diff
@@ -376,6 +379,119 @@ const char *get_loglevel_type_string(
 	return loglevel_type_string;
 }
 
+static
+int save_kernel_function_event(struct config_writer *writer,
+							   struct ltt_kernel_event *event)
+{
+	int ret;
+	ret = config_writer_open_element(writer, config_element_function_attributes);
+	if (ret) {
+		ret = LTTNG_ERR_SAVE_IO_FAIL;
+		goto end;
+	}
+
+	ret = config_writer_write_element_string(writer, config_element_name,
+		event->event->u.ftrace.symbol_name);
+	if (ret) {
+		ret = LTTNG_ERR_SAVE_IO_FAIL;
+		goto end;
+	}
+
+	/* /function attributes */
+	ret = config_writer_close_element(writer);
+	if (ret) {
+		ret = LTTNG_ERR_SAVE_IO_FAIL;
+		goto end;
+	}
+end:
+	return ret;
+
+}
+
+static
+int save_kernel_kprobe_event(struct config_writer *writer,
+			   struct ltt_kernel_event *event)
+{
+	int ret;
+	const char *symbol_name;
+	uint64_t addr;
+	uint64_t offset;
+
+	if (event->event->instrumentation == LTTNG_KERNEL_KPROBE) {
+		/*
+		 * Comments in lttng-kernel.h mention that
+		 * either addr or symbol_name are set, not both.
+		 */
+		addr = event->event->u.kprobe.addr;
+		offset = event->event->u.kprobe.offset;
+		symbol_name = addr ? NULL : event->event->u.kprobe.symbol_name;
+	} else {
+		symbol_name = event->event->u.kretprobe.symbol_name;
+		addr = event->event->u.kretprobe.addr;
+		offset = event->event->u.kretprobe.offset;
```

Comment:
Put the three assignments in the same order in the two blocks to make it easier to see the difference.